.github/workflows/main.yml

```diff
@@ -1,67 +1,71 @@
 name: Build
 
-on: [push]
+concurrency:
+    group: build-${{ github.head_ref }}
+    cancel-in-progress: true
+
+on:
+    push:
+        branches:
+            - master
```

Comment:
Why do we launch it only with pushes to master?

Reply:
With the last setup every workflow was running twice but only when those pull request were opened from this repo only. So the idea is, either you open a PR from a fork or this repo itself. But in either case you will be opening a PR to add changes, so exactly one build will run. And a final build will run when you merge the pull request.

Reply:
Also, now we wil be able to run it on demand too with workflow dispatch feature. So if nedded you can always trigger build for a different branch.